frontend/components/auction/collateral/MarketPriceSelection.vue

```diff
@@ -0,0 +1,117 @@
+                            <td class="pr-2 whitespace-nowrap">
+                                <span v-for="currency in callee[1].route" :key="currency"
+                                    >{{ currency }} &#8594;
+                                </span>
+                                DAI
+                            </td>
+                            <td class="w-full text-right whitespace-nowrap">
+                                <div v-if="callee[1].marketUnitPrice && !callee[1].marketUnitPrice.isNaN()">
+                                    <button type="button" @click="$emit('update:marketId', callee[0])">
+                                        <span v-if="marketId === callee[0]" class="opacity-50">Selected</span>
+                                        <span v-else class="text-green-500">Select</span>
+                                    </button>
+                                    <span class="pl-1">
+                                        <FormatCurrency :value="callee[1].marketUnitPrice" currency="DAI" /> per
+                                        <span class="uppercase">{{ auctionTransaction.collateralSymbol }}</span>
+                                    </span>
+                                </div>
+                                <div v-else class="opacity-50">Unknown</div>
+                            </td>
+                        </tr>
+                    </tbody>
+                </table>
+            </div>
+        </CollapseTransition>
+    </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import BigNumber from 'bignumber.js';
+import { Icon } from 'ant-design-vue';
+import CollapseTransition from '@ivanv/vue-collapse-transition';
+import { AuctionTransaction } from 'auctions-core/src/types';
+import FormatCurrency from '~/components/common/formatters/FormatCurrency.vue';
+
+export default Vue.extend({
+    components: {
+        Icon,
+        FormatCurrency,
+        CollapseTransition,
+    },
+    props: {
+        auctionTransaction: {
+            type: Object as Vue.PropType<AuctionTransaction>,
+            required: true,
+        },
+        marketId: {
+            type: String,
+            default: '',
+        },
+    },
+    data() {
+        return {
+            isExpanded: false,
+        };
+    },
+    computed: {
+        suggestionOrSelection(): string | undefined {
+            return this.marketId || this.auctionTransaction.suggestedMarketId;
+        },
+        marketUnitPrice(): BigNumber | undefined {
+            if (this.auctionTransaction.marketDataRecords && this.suggestionOrSelection) {
+                return this.auctionTransaction.marketDataRecords[this.suggestionOrSelection].marketUnitPrice;
+            }
+            return undefined;
+        },
+        callees() {
+            const sorted = [];
+            const unknown = [];
+            for (const callee in this.auctionTransaction.marketDataRecords) {
```

Comment:
How does this even pass typescript tests without the types? What exactly does this function returns? I can't figure it out from the names `callee`, `a`, `b`, `callee[0]`, `callee[1]`

Didn't you already have better (readable) logic implemented in the `getBestMarketId`?